test/tchannel_bridge.js

```diff
@@ -0,0 +1,157 @@
+        { description: 'tchannelAsThrift', channelEncoding: TChannelAsThrift }
+    ];
+
+    _.each(options, (o) => {
+
+        it (o.description + ' spans propagate through tchannel and preserve parent span properties', function tchTest1(done) {
+            let span = tracer.startSpan('futurama');
+            span.setBaggageItem('leela', 'fry');
+            let headers = TChannelBridge.saveTracerStateToCarrier(span);
+            TChannelBridge.saveBaggageToCarrier(span, headers);
+            let server = new TChannel({
+                serviceName: 'server',
+                trace: true,
+                forceTrace: true
+            });
+            // Server calls client channel after it starts listening.
+            server.listen(4040, '127.0.0.1', onServerListening);
+
+
+            // Create the toplevel client channel.
+            let client = new TChannel({
+                trace: true,
+                forceTrace: true
+            });
+
+
+            // Create the client subchannel that makes requests.
+            let clientSubChannel = client.makeSubChannel({
+                serviceName: 'server',
+                peers: ['127.0.0.1:4040']
+            });
+
+            // Wrap the subchannel in an encoding
+            let encodedChannel = o.channelEncoding({
+                channel: clientSubChannel,
+                entryPoint: path.join(__dirname, 'thrift', 'echo.thrift') // ignored in json case
+            });
+
+            // register the server request function.
+            let context = {};
+            encodedChannel.register(server, 'Echo::echo', context, handleServerReq);
+            function handleServerReq(context, req, head, body, callback) {
+
+                callback(null, {
+                    ok: true,
+                    head: head,
+                    body: body
+                });
+            }
+
+            function onServerListening() {
+                let req = encodedChannel.request({
+                    serviceName: 'server',
+                    // We don't want tchannel to generate any spans.
+                    parent: {span: TChannelBridge.getTChannelParentSpan() },
+                    headers: { cn: 'echo' },
+                    timeout: BIG_TIMEOUT
+                });
+                clientSubChannel.trace = false;
+                server.trace = false;
+                req.trace = false;
+
+                req.send('Echo::echo', headers, { value: 'some-string' }, (err, res, arg2, arg3) => {
+                    assert.isNotOk(err);
+                    let receivedSpan = TChannelBridge.getSpanFromTChannelRequest(tracer, 'nibbler', res.head);
+                    assert.equal(receivedSpan.name, 'nibbler');
+                    assert.equal(span.context().traceIdStr, receivedSpan.context().traceIdStr);
+
+                    let headers = res.head[constants.TCHANNEL_TRACING_PREFIX + constants.TRACER_STATE_HEADER_NAME].split(':');
```

Comment:
I want to verify that the headers contain the original span, and not the child of the original span, which is what the context gets populated with  in the new version.